pkgs/by-name/li/libguytools/package.nix

```diff
@@ -0,0 +1,56 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  libsForQt5,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libguytools";
+  version = "2.1.0";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/project/libguytools/libguytools/LatestSource/tools-${finalAttrs.version}.tar.gz";
+    hash = "sha256-eVYvjo2wKW2g9/9hL9nbQa1FRWDMMqMHok0V/adPHVY=";
+  };
+
+  qmakeFlags = [
+    "trunk.pro"
+    "toolsstatic.pro"
+  ];
+
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [
+    libsForQt5.qmake
+    libsForQt5.wrapQtAppsHook
+  ];
+  dontWrapQtApps = true;
+  buildInputs = [ libsForQt5.qtbase ];
+
+  postPatch = ''
+    sed -i "/dpkg-buildflags/d" tools.pro
+    patchShebangs create_version_file.sh
+  '';
+
+  preConfigure = ''
+    ./create_version_file.sh
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    cp -r lib $out/
+    cp -r include $out/
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Small programming toolbox";
+    mainProgram = "libguytools";
+    homepage = "https://libguytools.sourceforge.io";
+    maintainers = with lib.maintainers; [ d3vil0p3r ];
+    platforms = lib.platforms.unix;
```

Comment:
https://github.com/NixOS/nixpkgs/pull/278529/checks?check_run_id=32689131906

is darwin supported ?

Reply:
I don't have an endpoint where to actually test it.

Reply:
The build failure is just due to GHC not being present in the cache. (and it doesn't build in the timeout for PRs)

Reply:
How can we proceed?

Reply:
Either wait until GHC is built successfully, merge anyway or find someone with a darwin builder.

Reply:
looks like its broken on darwin anyway

```
error: builder for '/nix/store/sn4lpclhdid87q30g0dlyvbwjkr0a3ns-libguytools-2.1.0.drv' failed with exit code 2;
       last 25 log lines:
       >       (*ToolSignalLocal.pLogFn)(Error, pthread_self(), pFileName, pFunctionName, LineNr, pFormat, VaList);
       >                                        ^~~~~~~~~~~~~~
       > toolsignal.cpp:271:20: error: incompatible pointer to integer conversion assigning to 'unsigned long long' from 'pthread_t _Nonnull' (aka '_opaque_pthread_t *')
       >       *pThreadId = pthread_self();
       >                    ^~~~~~~~~~~~~~
       > 2 errors generated.
       > make: *** [Makefile:421: toolsignal.o] Error 1
       > make: *** Waiting for unfinished jobs....
       > toolsysinfo.cpp:40:10: fatal error: 'linux/if.h' file not found
       > #include <linux/if.h>
       >          ^~~~~~~~~~~~
       > toollog.cpp:116:87: error: cast from pointer to smaller type 'unsigned int' loses information
       >       wr += (size_t) snprintf (&LogLineHeader[wr], sizeof(LogLineHeader)-wr, "%08X ", (unsigned int)pthread_self());
       >                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > toollog.cpp:130:18: error: use of undeclared identifier 'fopen64'; did you mean 'fopen'?
       >          pFile = fopen64 (pLogFilename, "a");
       >                  ^~~~~~~
       >                  fopen
       > /nix/store/lq1j6jzsd0av6cm3yd9fxkjbw1mj2991-apple-sdk-10.14.6/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:153:7: note: 'fopen' declared here
       > FILE    *fopen(const char * __restrict __filename, const char * __restrict __mode) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fopen));
       >          ^
       > 1 error generated.
       > make: *** [Makefile:415: toolsysinfo.o] Error 1
       > 2 errors generated.
       > make: *** [Makefile:404: toollog.o] Error 1
       For full logs, run 'nix-store -l /nix/store/sn4lpclhdid87q30g0dlyvbwjkr0a3ns-libguytools-2.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/cpccgqf4yzaihq4v0bmmc98icdalyr55-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/278529

--------- Report for 'x86_64-darwin' ---------
1 package marked as broken and skipped:
guymager

1 package failed to build:
libguytools
```

maybe easier to just mark is as broken on darwin, file an issue upstream

Reply:
Issue reported to the guymager developer